cinema/serializers.py

```
@@ -1,13 +1,69 @@
 from rest_framework import serializers
 
-from cinema.models import Movie
+from cinema.models import Movie, Actor, Genre, CinemaHall
+
+
+class ActorSerializer(serializers.Serializer):
+    id = serializers.IntegerField(read_only=True)
+    first_name = serializers.CharField(max_length=100)
+    last_name = serializers.CharField(max_length=100)
+
+    def create(self, validated_data):
+        return Actor.objects.create(**validated_data)
+
+    def update(self, instance, validated_data):
+        instance.first_name = validated_data.get(
+            "first_name",
+            instance.first_name
+        )
+        instance.last_name = validated_data.get(
+            "last_name",
+            instance.last_name
+        )
+        instance.save()
+        return instance
+
+
+class GenreSerializer(serializers.Serializer):
+    id = serializers.IntegerField(read_only=True)
+    name = serializers.CharField(max_length=100)
```

The `name` field is declared but the serializer does not enforce model-level uniqueness. Checklist requirement: "Genre.name must be unique at the database/model level (unique=True) to support fixture load and validation." As implemented, creating a duplicate name will call `Genre.objects.create(...)` and likely raise a DB IntegrityError (500) instead of returning 400 with serializer.errors. Fix by using a ModelSerializer (recommended) or adding a UniqueValidator on this field so duplicates produce validation errors (400).